### PR TITLE
Add sha256sum to dev-env

### PR DESCRIPTION
**Pull request by cocreature**
Not quite sure what changed but our macos builds are unhappy and this
fixes it in my tests.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: nix/default.nix [hunk 1]
```diff
@@ -178,6 +178,7 @@ in rec {
     jq = bazel_dependencies.jq;
     sed = pkgs.gnused;
     sha1sum = pkgs.coreutils;
+    sha256sum = pkgs.coreutils;
```
Comment (garyverhaegen-da) [inline nix/default.nix:181]: That makes a ton of sense as a reason for things not to work, but it's very confusing as a reason for things to _stop working now_.

Comment (cocreature) [reply nix/default.nix:181]: yeah I don’t understand it but given that this fixes it in my tests, I’m also not that keen in spending more time debugging it. Maybe someone updated the macos builders or osmething like that.

**Review by garyverhaegen-da — APPROVED** [review]
Thanks!

File: nix/default.nix (hunk 1 above)
Comment (garyverhaegen-da) [reply nix/default.nix:181]: I'll run a compat test on this commit to be sure.

**Comment by garyverhaegen-da** [comment]
As one might expect, the compat test is far from over, but the macOS build has at least succeeded in getting far beyond the point where it used to crash on this, so that's a success in my book. I don't think we need to wait for the compat test to finish to merge this.

**Pull request merged**